examples/viewer_lib/ui/segmentation/segment_editor_ui.py

```diff
@@ -22,142 +25,185 @@
 from .threshold_effect_ui import ThresholdEffectUI
 
 
 @dataclass
 class SegmentEditorState:
     segment_list: SegmentListState = field(default_factory=SegmentListState)
-    segment_opacity: SegmentOpacityState = field(default_factory=SegmentOpacityState)
+    segment_rendering: SegmentRenderingState = field(default_factory=SegmentRenderingState)
     can_undo: bool = False
     can_redo: bool = False
-    show_3d: bool = False
     active_effect_name: str = ""
 
 
-class SegmentEditorUI(VContainer):
+class SegmentEditorUI(AbstractToolUI):
     toggle_segment_visibility_clicked = Signal(str)
-    edit_segment_clicked = Signal(str)
+    edit_segment_color_clicked = Signal(str)
     delete_segment_clicked = Signal(str)
     select_segment_clicked = Signal(str)
     add_segment_clicked = Signal()
     effect_button_clicked = Signal(type[SegmentationEffect])
 
     undo_clicked = Signal()
     redo_clicked = Signal()
-    show_3d_clicked = Signal()
 
     def __init__(self, **kwargs):
-        super().__init__(classes="pa-0 ma-0", **kwargs)
+        super().__init__(**kwargs)
         self._typed_state = TypedState(self.state, SegmentEditorState)
         self._effect_ui: dict[type[SegmentationEffect], Any] = {}
-
-        with self:
-            self.edit_dialog = SegmentEditDialog(TypedState(self.state, SegmentEditDialogState))
-
-            with VRow():
-                ControlButton(
-                    name="Undo",
-                    icon="mdi-undo",
-                    size=0,
-                    click=self.undo_clicked,
-                    disabled=(f"!{self._typed_state.name.can_undo}",),
-                )
-                ControlButton(
-                    name="Redo",
-                    icon="mdi-redo",
-                    size=0,
-                    style="margin-right: auto;",
-                    click=self.redo_clicked,
-                    disabled=(f"!{self._typed_state.name.can_redo}",),
-                )
-
-                ControlButton(
-                    name="Toggle 3D",
-                    icon="mdi-video-3d",
-                    size=0,
-                    click=self.show_3d_clicked,
-                    active=(f"{self._typed_state.name.show_3d}",),
-                )
-                VDivider()
-
-            with VRow(style="max-height: 300px; overflow-y: auto;"):
-                self._create_segment_list()
-
-            with VRow(classes="fill-width justify-center mb-4"):
-                with VTooltip(text="Add Segment"), Template(v_slot_activator="{ props }"):
-                    VBtn(
-                        v_bind="props",
-                        variant="tonal",
-                        density="compact",
-                        icon="mdi-plus",
-                        click=self.add_segment_clicked,
-                    )
+        self.edit_ui = SegmentEditUI()
+
+    @property
+    def name(self):
+        return "segment_editor"
+
+    def build_drawer_ui(self, **kwargs):
```

Comment:
Not a fan of this abstract class inheritance.

Previous implementation was more straight forward in my opinion and preserved better each abstraction levels and single responsibility principles.

SegmentEditorUI should be a pure UI component that you can instantiate anywhere and pass a v_if if needed.
UI building should be done in the CTor to keep RAII principles.

Buttons component should be added to the nav bar independently.